lib/Dialect/FIRRTL/Transforms/IMDeadCodeElim.cpp

```diff
@@ -287,8 +381,28 @@ void IMDeadCodeElimPass::runOnOperation() {
 
   // If a value changed liveness then propagate liveness through its users and
   // definition.
-  while (!worklist.empty())
-    visitValue(worklist.pop_back_val());
+  while (!valueWorklist.empty())
+    visitValue(valueWorklist.pop_back_val());
+
+  // Clean up annotations.
+  for (auto module : circuit.getBodyBlock()->getOps<FModuleOp>()) {
+    auto filter = [&](int _, Annotation anno) {
+      auto hierPathSym = anno.getMember<FlatSymbolRefAttr>("circt.nonlocal");
+      if (!hierPathSym)
+        return false;
+      auto op =
+          symbolTable->template lookup<hw::HierPathOp>(hierPathSym.getAttr());
```

Comment:
I seem to recall something about this `template` usage tripping up MSVC, but I'm not sure.

Reply:
Thanks, testing here https://github.com/llvm/circt/actions/runs/5070937958